tests/material/test_ncpflash.cpp

```diff
@@ -177,102 +171,154 @@ inline void testAll()
     enum { H2OIdx = FluidSystem::H2OIdx };
     enum { N2Idx = FluidSystem::N2Idx };
 
-    typedef Opm::TwoPhaseMaterialTraits<Scalar, liquidPhaseIdx, gasPhaseIdx> MaterialTraits;
-    typedef Opm::RegularizedBrooksCorey<MaterialTraits> EffMaterialLaw;
-    typedef Opm::EffToAbsLaw<EffMaterialLaw> MaterialLaw;
-    typedef typename MaterialLaw::Params MaterialLawParams;
+    using MaterialLawTraits = Opm::TwoPhaseMaterialTraits<Scalar, liquidPhaseIdx, gasPhaseIdx>;
+    using EffMaterialLaw = Opm::RegularizedBrooksCorey<MaterialLawTraits>;
+    using MaterialLaw = Opm::EffToAbsLaw<EffMaterialLaw>;
+    using MaterialLawParams = typename MaterialLaw::Params;
 
-    std::cout << "---- using " << Dune::className<Scalar>() << " as scalar ----\n";
-    Scalar T = 273.15 + 25;
+    Fixture()
+    {
+        Scalar T = 273.15 + 25;
 
-    // initialize the tables of the fluid system
-    Scalar Tmin = T - 1.0;
-    Scalar Tmax = T + 1.0;
-    unsigned nT = 3;
+        // initialize the tables of the fluid system
+        Scalar Tmin = T - 1.0;
+        Scalar Tmax = T + 1.0;
+        unsigned nT = 3;
 
-    Scalar pmin = 0.0;
-    Scalar pmax = 1.25 * 2e6;
-    unsigned np = 100;
+        Scalar pmin = 0.0;
+        Scalar pmax = 1.25 * 2e6;
+        unsigned np = 100;
 
-    FluidSystem::init(Tmin, Tmax, nT, pmin, pmax, np);
+        FluidSystem::init(Tmin, Tmax, nT, pmin, pmax, np);
 
-    // set the parameters for the capillary pressure law
-    MaterialLawParams matParams;
-    matParams.setResidualSaturation(MaterialLaw::wettingPhaseIdx, 0.0);
-    matParams.setResidualSaturation(MaterialLaw::nonWettingPhaseIdx, 0.0);
-    matParams.setEntryPressure(0);
-    matParams.setLambda(2.0);
-    matParams.finalize();
+        // set the parameters for the capillary pressure law
+        matParams.setResidualSaturation(MaterialLaw::wettingPhaseIdx, 0.0);
+        matParams.setResidualSaturation(MaterialLaw::nonWettingPhaseIdx, 0.0);
+        matParams.setEntryPressure(0);
+        matParams.setLambda(2.0);
+        matParams.finalize();
+
+        // create an fluid state which is consistent
+
+        // set the fluid temperatures
+        fsRef.setTemperature(T);
+    }
+
+    static Fixture& getInstance()
+    {
+        static std::unique_ptr<Fixture> instance;
+        if (!instance) {
+            instance = std::make_unique<Fixture>();
+        }
```

Comment:
Why do we use a pointer here?

Reply:
probably not a real need, but i wanted to put it on the heap, not the bss.

Reply:
sure.